programs/drift/src/controller/insurance.rs

```diff
@@ -677,11 +678,13 @@ pub fn settle_revenue_to_insurance_fund(
     )?
     .cast::<u64>()?;
 
-    validate!(
-        insurance_fund_token_amount != 0,
-        ErrorCode::NoRevenueToSettleToIF,
-        "no amount to settle to insurance fund"
-    )?;
+    if check_invariants {
+        validate!(
+            insurance_fund_token_amount != 0,
+            ErrorCode::NoRevenueToSettleToIF,
+            "no amount to settle to insurance fund"
+        )?;
+    }
```

Comment:
might wanna just return in the else case (assuming that a 0 amount is essentially a no-op and everything else below is just extra compute)

Reply:
or maybe not, that `InsuranceFundRecord` could be useful even for a 0 settle

Reply:
no it might make sense to just not emit/update any state in that case